tools/test/export/build_test.py

```diff
@@ -16,154 +16,165 @@
-
-
-class ProgenBuildTest():
-    def __init__(self, desired_ides, targets):
-        #map of targets and the ides that can build programs for them
-        self.target_ides = {}
-        for target in targets:
-            self.target_ides[target] =[]
-            for ide in desired_ides:
-                if target in EXPORTERS[ide].TARGETS:
-                    #target is supported by ide
-                    self.target_ides[target].append(ide)
-            if len(self.target_ides[target]) == 0:
-                del self.target_ides[target]
-
-
-    @staticmethod
-    def get_pgen_targets(ides):
-        #targets supported by pgen and desired ides for tests
-        targs = []
-        for ide in ides:
-            for target in TARGET_NAMES:
-                if target not in targs and hasattr(TARGET_MAP[target],'progen') \
-                        and ProGenDef(ide).is_supported(TARGET_MAP[target].progen['target']):
-                    targs.append(target)
-        return targs
+from tools.targets import TARGET_NAMES
+from tools.tests import TESTS
+from tools.project import setup_project
+from tools.project_api import print_results, export_project
+from tools.tests import test_name_known, Test
+from tools.export.exporters import FailedBuildException, \
+                                   TargetNotSupportedException
+from tools.utils import argparse_force_lowercase_type, \
+                        argparse_force_uppercase_type, argparse_many
+
+
+class ProgenBuildTest(object):
+    """Object to encapsulate logic for progen build testing"""
+    def __init__(self, desired_ides, mcus, tests):
+        """
+        Initialize an instance of class ProgenBuildTest
+        Args:
+            desired_ides: the IDEs you wish to make/build project files for
+            mcus: the mcus to specify in project files
+            tests: the test projects to make/build project files from
+        """
+        self.ides = desired_ides
+        self.mcus = mcus
+        self.tests = tests
+
+    @property
+    def mcu_ide_pairs(self):
+        """Yields tuples of valid mcu, ide combinations"""
+        for mcu in self.mcus:
+            for ide in self.ides:
+                if mcu in EXPORTERS[ide].TARGETS:
+                    yield mcu, ide
 
     @staticmethod
-    def handle_project_files(project_dir, mcu, test, tool, clean=False):
+    def handle_log_files(project_dir, tool, name):
+        """
+        Renames/moves log files
+        Args:
+            project_dir: the directory that contains project files
+            tool: the ide that created the project files
+            name: the name of the project
+            clean: a boolean value determining whether to remove the
+                   created project files
+        """
         log = ''
         if tool == 'uvision' or tool == 'uvision5':
-            log = os.path.join(project_dir,"build","build_log.txt")
+            log = path.join(project_dir, "build", "build_log.txt")
```

Comment:
It is just the way the build command is called within progen. After this is merged, I can make a PR to change that.  Either they can both be at the same level as the project file, or, as uvision, be written in a build directory.